scripts/coverage_report_jenkins.py

```diff
@@ -374,106 +386,215 @@ def sonar_scan(ssh, project_name, project_version, scanner_url, scanner_dir, ser
+        # not the case this is not an eligible build.
         if 'appliance_version' not in artifacts:
-            logger.info('appliance_version not in artifacts of %s/%s', job_name, build_number)
+            logger.info('appliance_version not in artifacts of %s/%s', jenkins_job, build_number)
             continue
-
         build_appliance_version = download_artifact(
-            jenkins_user, jenkins_token, jenkins_url, job_name, build_number,
+            jenkins_data.user,
+            jenkins_data.token,
+            jenkins_data.url,
+            jenkins_job,
+            build_number,
             artifacts['appliance_version']['relativePath']).strip()
-
         if not build_appliance_version:
             logger.info('Appliance version unspecified for build %s', build_number)
             continue
 
-        if Version(build_appliance_version) < Version(appliance_version):
+        # Build versions that are less than the target version are invalid
+        if Version(build_appliance_version) < Version(cfme_version):
             logger.info(
-                'Build %s already has lower version (%s)', build_number, build_appliance_version)
+                'Build %s already has lower version (%s) than target version (%s)',
+                build_number, build_appliance_version, cfme_version)
             logger.info('Ending here')
             break
 
+        # We must have the actual coverage data tarball in the artifacts.
+        # If we do set it in our build object.
         if 'coverage-results.tgz' not in artifacts:
-            logger.info('coverage-results.tgz not in artifacts of %s/%s', job_name, build_number)
+            logger.info('coverage-results.tgz not in artifacts of %s/%s', jenkins_job, build_number)
             continue
 
+        # We have all the data to instantiate our Build object, so lets do it.
+        # Note, we could fill out its data members a little at a time because
+        # a namedtuple's data members are immutable.
+        build = Build(
+            number=build_number,
+            job=jenkins_job,
+            coverage_archive=artifacts['coverage-results.tgz']['relativePath'])
+
         if not check_artifact(
-                jenkins_user, jenkins_token, jenkins_url, job_name, build_number,
+                jenkins_data.user,
+                jenkins_data.token,
+                jenkins_data.url,
+                jenkins_job,
+                build_number,
                 artifacts['coverage-results.tgz']['relativePath']):
-            logger.info('Coverage archive not possible to be downloaded, skipping')
+            logger.info('Coverage archive could not possibly be downloaded, skipping')
             continue
 
-        if build_appliance_version == appliance_version:
+        if build_appliance_version == cfme_version:
             logger.info('Build %s was found to contain what is needed', build_number)
```

Comment:
**Optional**

Actually log `build` object here, instead of just the number, for complete info about the build.